### PR TITLE
Integrate LLVM at llvm/llvm-project@c21928285430

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Update BUILD files and submodule for
[c21928285430](https://github.com/llvm/llvm-project/commit/c21928285430)

### CODE REVIEW

File: llvm-bazel/llvm_configs/abi-breaking.h.cmake [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 /* Allow selectively disabling link-time mismatch checking so that header-only
    ADT content from LLVM can be used without linking libSupport. */
-#if !LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING
+#if !defined(LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING) || !LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING
```
Comment (chandlerc) [inline llvm-bazel/llvm_configs/abi-breaking.h.cmake:23]: Do we not need to update our `abi-breaking.h` file?

**Review by chandlerc — CHANGES_REQUESTED** [review]

File: llvm-bazel/llvm_configs/abi-breaking.h.cmake (hunk 1 above)
Comment (GMNGeoffrey) [reply llvm-bazel/llvm_configs/abi-breaking.h.cmake:23]: We don't have one. We just have a Bazel rule that depends on the specific structure (but not this part of it)

https://github.com/google/llvm-bazel/blob/main/llvm-bazel/llvm-project-overlay/llvm/BUILD#L113-L124

Comment (chandlerc) [reply llvm-bazel/llvm_configs/abi-breaking.h.cmake:23]: Doh, sorry, forgot this one worked differently. Should have checked.

(Also sorry I didn't see your response earlier.)

**Review by chandlerc — APPROVED** [review]
LGTM

File: llvm-bazel/llvm_configs/abi-breaking.h.cmake (hunk 1 above)
Comment (GMNGeoffrey) [reply llvm-bazel/llvm_configs/abi-breaking.h.cmake:23]: No worries

**Pull request merged**